android-components/components/feature/push/src/main/java/mozilla/components/feature/push/AutoPushFeature.kt

```diff
@@ -127,50 +138,38 @@ class AutoPushFeature(
      * This should only be done on an account logout or app data deletion.
      */
     override fun shutdown() {
-        connection.ifInitialized {
-            coroutineScope.launch {
-                unsubscribeAll()
-            }
+        withConnection {
+            it.unsubscribeAll()
         }
-
-        // Reset the push subscription check.
-        prefLastVerified = 0L
     }
 
     /**
      * New registration tokens are received and sent to the AutoPush server which also performs subscriptions for
      * each push type and notifies the subscribers.
      */
     override fun onNewToken(newToken: String) {
-        coroutineScope.launchAndTry {
+        val currentConnection = connection
+        coroutineScope.launch {
             logger.info("Received a new registration token from push service.")
 
             saveToken(context, newToken)
 
             // Tell the autopush service about it and update subscriptions.
-            connection.updateToken(newToken)
-            tryVerifySubscriptions()
+            currentConnection?.update(newToken)
+            verifyActiveSubscriptions()
```

Comment:
Aaah the reason it doesn't is that we still want to call `saveToken` even if the connection isn't open yet, the `update` call in that case should just be a no-op and went he client is opened, it will call `update`